src/neo/SmartContract/Native/NeoToken.cs

```diff
@@ -344,7 +344,26 @@ private async ContractTask<bool> Vote(ApplicationEngine engine, UInt160 account,
             (
                 p.Key.Key.AsSerializable<ECPoint>(1),
                 p.Value.GetInteroperable<CandidateState>()
-            )).Where(p => p.Item2.Registered).Select(p => (p.Item1, p.Item2.Votes)).ToArray();
+            )).Where(p => p.Item2.Registered).Select(p => (p.Item1, p.Item2.Votes)).Take(100).ToArray();
```

Comment:
How does the caller retrieve the remaining candidates if there are more than 100?

Reply:
Do you think that it's needed?

Reply:
This method was used by Grant Shares so or we add a new method for paginate it, or I think that we can't add a new arguments here (without versioning)

Reply:
> Do you think that it's needed?

Yes, I think it's a bad idea to arbitrarily limit return values without providing a mechanism to paginate. We have similar issues w/ accessing iterators over JSON-RPC (https://github.com/neo-project/neo-modules/issues/629 and https://github.com/neo-project/neo-devpack-dotnet/issues/647)

> This method was used by Grant Shares so or we add a new method for paginate it, or I think that we can't add a new arguments here (without versioning)

We can add an overload that takes skip + take parameters and returns a structure that has the `(ECPoint PublicKey, BigInteger Votes)` tuple array + a boolean to indicate if there are more results (similar to the [`FindState` State Service RPC method](https://docs.neo.org/docs/en-us/reference/rpc/latest-version/api/findstates.html)). That way we don't break existing callers but we can still provide the full functionality to contracts that switch to the new overload.

Reply:
Please, take a look again

Reply:
This works, though I like the [`FindState`](https://docs.neo.org/docs/en-us/reference/rpc/latest-version/api/findstates.html) pattern much better. 

I decided to build a single GitHub issue to track the Iterator JSON RPC return issue #2699. I actually think it would be pretty straightforward to add `Iterator` skip support. Assuming we can add that, then we could change GetCandidates to return an `Iterator`